v2/lint/base.go

```diff
@@ -83,14 +83,19 @@ func (l *Lint) CheckEffective(c *x509.Certificate) bool {
 // CheckEffective()
 // Execute()
 func (l *Lint) Execute(cert *x509.Certificate) *LintResult {
-	if l.Source == CABFBaselineRequirements && !util.IsServerAuthCert(cert) {
+	if l.Source == CABFBaselineRequirements && !util.IsServerAuthCert(cert) && !util.IsDelegatedOCSPResponderCert(cert) {
```

Comment:
@sleevi @cpu @zakird The more I think about this line, I wonder if we want to delete it in total and replace it with checks in the `CheckApplies` function of every lint. This would also support the upcoming SMIME-BRGs and the drafted Code Signing BRGs, especially since they will share a lot of requirements. Otherwise we will have more and more exceptions and strange combinations of OID checks.

Reply:
I don’t support doing that is this change, but I also think we should hold off on that refactoring until we get to that point. That seems like a lot of churn for unclear benefit, at least at present, although I can agree with you in the hypothetical.

One of the issues we had with QWACs is that it’s not merely sufficient for something to be a certificate; lints need an invested community of maintainers who are familiar with the specifications, the interpretations, and the use case. I suspect few, if any, of the maintainers are interested in Code Signing. The S/MIME WG activity seems too early to tell if it will produce something useful for Internet mail clients, given that right now it appears to be narrowly focusing on Enterprise/Gov’t cases and/or letting them dominate the use cases. As such, it may end up that those Guidelines aren’t adopted by major mail clients, or their audits not accepted/not seen as relevant, if they don’t end up being useful, and that seems like it could similarly impact the ability to support or maintain lints.

Given that historically ZLint has exclusively focused on TLS certs, I think the broader meta-question about other profiles is one we should hold off on, until we’ve got a body of lints and active maintainer interest. In the worst case, the structure of lints was designed to facilitate mixins of other lints in the interim, and wouldn’t be (negatively) affected by the current checks.

Reply:
I'll split an issue based on this conversation, so we don't forget about it and close this conversation here.

Reply:
IMO it is clearly under zlint's purview to lint any industry regulated certificate type, but totally agree that it requires invested maintainers to make it valuable. As other cert types become more useful for various purposes, one would hope that invested CAs would desire to contribute to the project (eventually) 🙂